### PR TITLE
 bounty: add report to funder

### PR DESCRIPTION
**Pull request by thelostone-mc**
##### Description

- [x] Adds ability for a contributor to notify funder on how to make bounty more appealing via popup
- [x] Send Email to funder
- [x] Evaluate and see if this data needs to be persisted

How it works ->

_**FE flow:**_ 
Logged in User -> navigates to issue -> clicks on report -> selects reporting reason -> clicks submit
-> gets a toaster notification 

_**BE Flow**_
- Upon clicking Report -> we date NotifyFunder tracking who reported what on which bounty which `funder_notified` = `False`
- When an email is ready to be sent out -> we query all bounties for a funder where `funder_notified` = `False`  -> Send an email and update the `funder_notified` = `True` ensuring dont' send duplicates

@mbeacom / @SaptakS 
- Could you just wire up when the mail sending trigger should be done
- How do you handle when there are no notifications to be sent out ?  I'm just returning [None](https://github.com/gitcoinco/web/pull/2245/files#diff-5af8ba1f144445e3932163d9adb42fa7R542)
 
Bonus

- [x] Added toaster component (aka small alert which appear and disappear on it's own after a few seconds) 

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Testing

x

##### Refers/Fixes

- https://github.com/gitcoinco/web/issues/2219

### CODE REVIEW

File: app/retail/emails.py [hunk 1]
```diff
@@ -521,6 +521,50 @@ def render_start_work_applicant_expired(interest, bounty):
...
+    from django.db.models import Count
+    to_email = profile.email
+
+    # Fetch All Bounties which have entry in NotifyFunder &
+    # NotifyFunder.funder_notified == false &
+    # Calculate count for each notify_reason for a bounty
+
+    bounties = Bounty.objects.filter(bounty_owner_github_username=profile.handle)
+    for bounty in bounties:
+        notifs = NotifyFunder.objects.filter(bounty=bounty.pk).values('notify_reasons').annotate(Count('notify_reasons'))
```
Comment (mbeacom) [inline app/retail/emails.py]: @thelostone-mc `NotifyFunder.objects.filter(bounty=bounty.pk).annotate(arr_els=Func(F('notify_reasons'), function='unnest')).values('notify_reasons').order_by('notify_reasons').annotate(counterz=Count('notify_reasons'))`
 ?

File: app/retail/emails.py [hunk 2]
```diff
@@ -528,10 +528,11 @@ def render_notify_funder(profile):
     to_email = profile.email
 
     bounties = Bounty.objects.filter(bounty_owner_github_username=profile.handle)
-    notifications =[]
+    notifications = []
     for bounty in bounties:
         notifs_list = NotifyFunder.objects.filter(bounty=bounty.pk, funder_notified=False)
-        notifs = notifs_list.annotate(arr_els=Func(F('notify_reasons'), function='unnest')).values_list('arr_els', flat=True)
+        notifs = notifs_list.annotate(arr_els=Func(F('notify_reasons'), function='unnest')).\
```
Comment (mbeacom) [inline app/retail/emails.py]: Can you line break before the `.` ? So it's similar to:

```python
notifs = notifs_list.annotate(arr_els=Func(F('notify_reasons'), function='unnest')) \
    .values_list('arr_els', flat=True)
```
😛

File: app/retail/emails.py [hunk 3]
```diff
@@ -521,6 +521,40 @@ def render_start_work_applicant_expired(interest, bounty):
     return response_html, response_txt, subject
 
 
+def render_notify_funder(profile):
+    from dashboard.models import Bounty, NotifyFunder
+    from django.db.models import F, Func
+    from collections import Counter
```
Comment (mbeacom) [inline app/retail/emails.py]: 😁 Aye, can you move these imports to the top of the file?

Comment (thelostone-mc) [reply app/retail/emails.py]: will do ^_^

**Comment by thelostone-mc** [comment]
@mbeacom addressed ^_^

**Comment by thelostone-mc** [comment]
@PixelantDesign could we decide on how often this mail goes out ? If that's done -> we can code it up -> test it on staging -> merge it in

**Comment by owocki** [comment]
looks really good…  what does the email to the funder look like after the form is submitted.

 my only  worry is what beylin says here => https://twitter.com/MarkBeylin/status/1049668569246720000

 i wonder if we need some input for a freeform comment in there too.

**Comment by PixelantDesign** [comment]
Need to configure when the mail gets sent out.

**Pull request closed without merging**